### ISSUE TITLE
[strict-boolean-expressions] Type guards/predicates are treated as any when imported from aliased path

### ISSUE THREAD

**Issue by zalanlevai** [typescript-eslint/typescript-eslint#2327]
**Repro**

##### .eslintrc
```json
{
    "rules": {
        "@typescript-eslint/strict-boolean-expressions": ["error"],
    }
}
```

##### tsconfig.json
```json
{
    "compilerOptions": {
        "baseUrl": "src",
        "paths": {
            "@/*": [ "./*" ],
        }
    }
}
```

##### src/foo.ts
```ts
export interface Foo {
    prop: string;
}

export function isFoo(obj: unknown): obj is Foo {
    const unsafeObj = obj as any;
    return unsafeObj.prop != null && typeof unsafeObj.prop === "string";
}
```

##### src/test.ts
```ts
import { isFoo } from "./foo"; // Works as expected
import { isFoo } from "@/foo"; // Causes issue with type guard

declare const obj: unknown;

if (isFoo(obj)) {
    console.log("obj is Foo");
}
```

**Expected Result**

The type guard `isFoo` imported from the aliased path `@/foo` works exactly the same as the one imported from the unaliased path `./foo`.

**Actual Result**

The type guard `isFoo` imported from `@/foo` is treated as a function returning `any`, thus causing the following, incorrect error to be reported. This occurs both when running `eslint` through `vscode` and when running `eslint` directly.

![image](https://user-images.githubusercontent.com/12102131/88481130-edfc5f00-cf59-11ea-8700-4183de43f7e4.png)

**Additional Info**

The path alias is correctly resolved by TypeScript directly. This is reflected in type-checking output and in `vscode` as well.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.0` |
| `@typescript-eslint/parser`        | `3.6.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.5.0` |
| `node`                             | `14.5.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
The rule just consumes the types from typescript - so if typescript reports `any` for one case but not the other, then it's likely a configuration issue on your end.

You can try running your linter with `DEBUG=typescript-eslint:eslint-plugin:*` (i.e. `DEBUG=typescript-eslint:eslint-plugin:* yarn eslint path/to/file.ts`).
I suspect you'll see a log line which says `Found an "error" any type`. This log line only shows up if TS cannot resolve the type (eg configuration issue).

**Comment by zalanlevai** [comment]
The issue was in fact caused by improper configuration on my end; missed the [monorepo configuration](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/MONOREPO.md) in the guide. The `project` parser setting was set to point to the root `tsconfig` file which didn't include the path aliases I was using. Changing that to point to the `tsconfig` of each package fixed the issue. This explains the discrepancy I had between running `tsc` manually and the TS instance used by ESLint.

Works great, thank you!
